File: backend/app/schemas/apartment.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ApartmentCreate(BaseModel):
    address: str
    building: str | None = None
    apartment_number: str | None = None
    city: str
    rooms: int = Field(ge=1, le=10)
    sqm: float = Field(ge=10, le=300)
    floor: int | None = None
    specifications: dict | None = None


class ApartmentUpdate(BaseModel):
    address: str | None = None
    building: str | None = None
    apartment_number: str | None = None
    city: str | None = None
    rooms: int | None = Field(default=None, ge=1, le=10)
    sqm: float | None = Field(default=None, ge=10, le=300)
    floor: int | None = None
    specifications: dict | None = None
    status: str | None = None
    thumbnail_url: str | None = None


class ApartmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    address: str
    building: str | None = None
    apartment_number: str | None = None
    city: str
    rooms: int
    sqm: float
    floor: int | None = None
    specifications: dict | None = None
    status: str
    thumbnail_url: str | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/apartment.ts
export interface Apartment {
  id: string;
  address: string;
  building: string | null;
  apartment_number: string | null;
  city: string;
  rooms: number;
  sqm: number;
  floor: number | null;
  specifications: Record<string, boolean> | null;
  status: 'vacant' | 'listed' | 'rented' | 'move-out';
  thumbnail_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface ApartmentCreate {
  address: string;
  building?: string;
  apartment_number?: string;
  city: string;
  rooms: number;
  sqm: number;
  floor?: number;
  specifications?: Record<string, boolean>;
}

export interface Photo {
  id: string;
  apartment_id: string;
  storage_url: string;
  room_type: string | null;
  photo_type: 'move-in' | 'move-out';
  uploaded_at: string;
}

export interface InventoryItem {
  id: string;
  apartment_id: string;
  room_type: string | null;
  item_type: string;
  condition_notes: string | null;
  photo_id: string | null;
  created_at: string;
}

export interface Listing {
  id: string;
  apartment_id: string;
  platform: string;
  title: string;
  description: string;
  amenities: Record<string, any> | null;
  price: number | null;
  rental_type: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface LeasePeriod {
  id: string;
  apartment_id: string;
  tenant_name: string | null;
  start_date: string;
  end_date: string;
  rental_type: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface Conversation {
  id: string;
  apartment_id: string;
  tenant_name: string | null;
  platform_source: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface Message {
  id: string;
  conversation_id: string;
  sender: string;
  message_text: string;
  timestamp: string;
}
